### ISSUE TITLE
 SYSCALL 3Bh

### ISSUE THREAD

**Issue by jmgk77** [ps2dev/ps2sdk#268]
Hi,

I cant find any reference to  syscall 3Bh void JoinThread()... 

I also couldnt find anything in https://psi-rockin.github.io/ps2tek/, neither in other homebrews

Anybody have any information about it?

**Comment by sp193** [comment]
This syscall was never implemented in the EE kernels that we have.
It appears among the symbols of some Sony SDKs, but it has no implementation and is undocumented. It could be that Sony once had more functions planned, but did not complete them by the PlayStation 2's launch in year 2000.

**Comment by jmgk77** [comment]
I see... I am working in a [SDL2 port for PS2](https://github.com/jmgk77/SDL2_PS2) and several other ports implement SDL_WaitThread with a call to system's JoinThread(), and i though i could do the same

You know if the JoinThread() code exists in the BIOS code? Maybe I could dissassembly it and get some info how to use it...

**Comment by uyjulian** [comment]
As mentioned, it has no implementation. That means it doesn't exist / is stubbed / is no-op and you won't find anything in the BIOS code.

**Comment by jmgk77** [comment]
ok, now i understand

thanks for the info

**Comment by sp193** [comment]
It's not really a BIOS, but a boot ROM. The EE kernel is copied from said ROM and booted.

I would think you have a few solutions. The best solution would depend on the usecase.
1. Create a semaphore and pass it to the thread, to be signalled upon completion. Your caller thread can wait on the semaphore.
2. The caller thread could use a flag to check on whether the thread has exited. Since this pattern usually involves a busy-loop, the waiting thread should have a lower priority than the thread. If the task is extremely short and is executed many times, then this could actually be more efficient than (1).

**Comment by jmgk77** [comment]
@sp193

i am thinking in doing something similar to 1, but using the return value of ReferThreadStatus(). If i got -1, i know the thread dont exists anymore

**Comment by fjtrujy** [comment]
> I see... I am working in a [SDL2 port for PS2](https://github.com/jmgk77/SDL2_PS2) and several other ports implement SDL_WaitThread with a call to system's JoinThread(), and i though i could do the same
> 
> You know if the JoinThread() code exists in the BIOS code? Maybe I could dissassembly it and get some info how to use it...

As you know I'm doing as well the SDL2 port, and there are easier ways to implement the `SDL_WaitThread`, don't worry 🤔

**Comment by jmgk77** [comment]
Yes, i'm ready to do the PRs. But while I wait, i'm finishing the parts already started
